Kernel/FileSystem/SysFS/Subsystems/Kernel/DeviceMajorNumberAllocations.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2024, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/JsonObjectSerializer.h>
+#include <Kernel/Devices/MajorNumberAllocation.h>
+#include <Kernel/FileSystem/SysFS/Subsystems/Kernel/DeviceMajorNumberAllocations.h>
+
+namespace Kernel {
+
+UNMAP_AFTER_INIT SysFSDeviceMajorNumberAllocations::SysFSDeviceMajorNumberAllocations(SysFSDirectory const& parent_directory, DeviceNodeType device_node_type)
+    : SysFSGlobalInformation(parent_directory)
+    , m_device_node_type(device_node_type)
+{
+    VERIFY(m_device_node_type == DeviceNodeType::Block || m_device_node_type == DeviceNodeType::Character);
+}
+
+StringView SysFSDeviceMajorNumberAllocations::name() const
+{
+    return m_device_node_type == DeviceNodeType::Character ? "chardev_major_allocs"sv : "blockdev_major_allocs"sv;
```

Comment:
Are these names standard? They look a bit awkward to me.

Reply:
This is not a standard. but nothing in `sysfs` is a standard because nobody ever set a standard on it.
If the name is not good, please suggest something else and we can discuss it :)

Reply:
I suppose it's easy to change later.